src/Common/src/TypeSystem/IL/Stubs/PInvokeMarshallingILEmitter.cs

```diff
@@ -100,6 +100,16 @@ private static bool IsBlittableStruct(TypeDesc type)
         {
             if (type.IsValueType)
             {
+                // Bool and Char are primitive types that are not simple types.
+                // Need to special case them because they have a field of the same
+                // type (primitive types are the only types allowed to do that).
+                // This would lead to recursion if not special cased.
+                TypeFlags category = type.Category;
+                if (category == TypeFlags.Boolean || category == TypeFlags.Char)
```

Comment:
We are making assumption here that the "simple type" is not going to be passed to IsBlittableStruct...

I have noticed that we are calling IsSimpleType and IsBlittableStruct together in most places, and the places where we are not are just oversight.

Would it make sense to merge IsSimpleType and IsBlittableStruct into one method, something like:

```
bool IsBlittableType(Type type)
{
    type = type.UnderlyingType;

    if (type.IsValueType)
    {
         if (type.IsPrimitive)
         {
              // All primitive types except char and Boolean are blitable
              TypeFlags category = type.Category;  
              if (category == TypeFlags.Boolean || category == TypeFlags.Char) 
                   return false;
              return true;
         }
         
         ... the rest of IsBlittableStruct ...
}
```